lib/index.js

```diff
@@ -92,7 +92,7 @@ internals.handler = function (route, handlerOptions) {
 
     return async function (request, h) {
 
-        const { uri, headers } = await settings.mapUri(request);
+        let { uri, headers, method } = await settings.mapUri(request);
```

Comment:
I'd rather keep it that way:
```suggestion
        const { uri, headers, method = request.method } = await settings.mapUri(request);
```

Reply:
The proposed change by @Marsup would break my existing code, that changes `request.method` in `mapUri()`.

Reply:
Wait, will it break? I assumed the it fetches the `request.method` before calling `mapUri()`, but that is probably not the case, since it needs the result to know whether to fetch the default.

Reply:
I'd say it's working as expected: https://runkit.com/embed/03jrtcm2e10x

Reply:
Great! Reminder to self: Get my morning cup of coffee, *before* reviewing code.